### ISSUE TITLE
PCNTLHeartbeatSender would be never triggered again when connection in writing status 

### ISSUE THREAD

**Issue by MaxwellZY** [php-amqplib/php-amqplib#865]
```php
private function registerListener($interval)
{
    pcntl_signal(SIGALRM, function () use ($interval) {
        if (!$this->connection || $this->connection->isWriting()) {
            return;
        }

        if (!$this->connection->isConnected()) {
            $this->unregister();
            return;
        }

        if (time() > ($this->connection->getLastActivity() + $interval)) {
            $this->connection->checkHeartBeat();
        }

        pcntl_alarm($interval);
    });
}
```
When connection is writing, should trigger the checkHeartBeat later but not just return, the following code is fixed

```php
private function registerListener($interval)
{
    pcntl_signal(SIGALRM, function () use ($interval) {
        if (!$this->connection) {
            return;
        }

        if (!$this->connection->isConnected()) {
            $this->unregister();
            return;
        }
        
        if ($this->connection->isWriting()) {
            pcntl_alarm($interval);
            return;
        }

        if (time() > ($this->connection->getLastActivity() + $interval)) {
            $this->connection->checkHeartBeat();
        }

        pcntl_alarm($interval);
    });
}
```

**Comment by ramunasd** [comment]
@laurynasgadl Could You check this? I bet it's quite easy to add another functional test and fix it.

**Comment by laurynasgadl** [comment]
@ramunasd sure, I'll look into it

**Comment by ramunasd** [comment]
will be fixed in 2.12.2
